main.go

```diff
@@ -0,0 +1,82 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	"cloud.google.com/go/pubsub"
+)
+
+func main() {
+
+	err := publishMessage("pubsub1", "testing1", data{
+		messageID:    "1",
+		requestBody:  "api request body",
+		responseBody: "api response body",
+	})
+	if err != nil {
+		fmt.Println(err)
+		return
+	}
+}
+
+type data struct {
+	messageID		string
+	requestBody		string
+	responseBody	string
+}
+
+// PublishMessage publishes messages to already created topics
+func publishMessage(projectID, topicID string, msg data) error {
+	ctx, cancel := context.WithTimeout(context.Background(), 30 * time.Second)
+	defer cancel()
+
+	os.Setenv("PUBSUB_EMULATOR_HOST", "localhost:8085")
+
+	client, err := pubsub.NewClient(ctx, projectID)
+	if err != nil {
+		fmt.Println(err)
+		return err
+	}
+	defer client.Close()
+
+	topic := client.Topic(topicID)
+	defer topic.Stop()
+
+	exists, err := topic.Exists(ctx)
+	if err != nil {
+		fmt.Println(err)
+		return err
+	}
+
+	if exists {
+		fmt.Println("topic already exist")
+	}
+	
+	if !exists {
+		_, err = client.CreateTopic(ctx, topicID)
+		fmt.Printf("%s created", topicID)
+		if err != nil {
+			fmt.Println(err)
+			return err
+		}
+	}
+
+	msgg := &pubsub.Message{
+		ID:              projectID,
+		Data:            []byte(fmt.Sprintf(msg.messageID, msg.requestBody, msg.responseBody)),
```

Comment:
You're basically concatenating the message id with the request and response body. This will be practically impossible to interpret, since we wont know where one ends and the other one starts.

Instead, pick an encoding format like json or msgpack or protobuf. And stick with them. For simplicity purposes, let's just use json here. 

Just marshal the msg struct as json, and use that as the data object.